static/js/util/validation.js

```diff
@@ -0,0 +1,115 @@
+import _ from 'lodash';
+
+let handleNestedValidation = (profile, keys, nestedKey) => {
+  let nestedFields = index => (
+    keys.map(key => [nestedKey, index, key])
+  );
+  return profile[nestedKey].map((v, i) => nestedFields(i)).reduce((a,b) => a.concat(b));
```

Comment:
The `concat` here in the `reduce` will cause a new list to be created for each iteration of `reduce`, though the number of lists here is likely to be small. Instead could you use `_.flatten`?